jupyter_core/paths.py

```diff
@@ -24,14 +24,15 @@
 # It is used by BSD to indicate hidden files.
 UF_HIDDEN = getattr(stat, 'UF_HIDDEN', 32768)
 
+JUPYTER_ENV_PRIORITY = os.environ.get('JUPYTER_ENV_PRIORITY', 'n').lower() not in ['no', 'n', 'false', 'off', '0', '0.0']
```

Comment:
Regarding the name, I think it should include `PATH` in some manner, otherwise its a bit ambiguous as readers might wonder 'what environment is being prioritized'.

Some suggestions:
`JUPYTER_PREFER_ENV_PATH`
`JUPYTER_ENV_PATH_PRIORITY`
`JUPYTER_PRIORITIZE_ENV_PATH`

or using `PYENV` over `ENV`, then `JUPYTER_PYENV_PRIORITY` isn't so ambiguous either.

A short comment might be helpful here as well.

Reply:
or `JUPYTER_PATH_PREFER_ENV` so it is sort of next to `JUPYTER_PATH`, which is related (though this also applies to `JUPYTER_CONFIG_PATH`, so perhaps `JUPYTER_PREFER_ENV`)?

I like PREFER over PRIORITY as it is a simpler word.

Reply:
`JUPYTER_PREFER_ENV`?

Reply:
On second thought, I like `JUPYTER_PREFER_ENV_PATH` best of the options suggested so far. `JUPYTER_PATH_*` makes it seem like it doesn't apply to jupyter config, whereas `JUPYTER_PREFER_ENV_PATH` makes it clearer that we are talking about the environment path (where it comes up) rather than Jupyter's path.

Reply:
Fixed in latest commit.